src/driving/drifty_car.py

```diff
@@ -0,0 +1,230 @@
+    drive_mode.reverse_acceleration = drive_mode.forward_acceleration
+    return drive_mode
+
+
+class DriftyCar(DriveMode):
+    """
+    Driving mode with peppy acceleration, braking, turning, and a drifting
+    mechanic triggered by holding gas and brake at the same time.
+    """
+
+    def setup(self):
+        self.velocity = (0.0, 0.0)
+        # TODO
+        self._sprite = self.player.sprite
+
+        # Single point on the curve means it's a flat line: all Y values
+        # are the same
+        self.forward_acceleration = Curve([(0.0, 1500)])
+        """
+        Effectively, torque of the engine driving forward.
+        Since there is no brake, this is also braking power when driving in reverse and attempting to slow down.
+        """
+        self.forward_drifting_acceleration = Curve([(0.0, 500)])
+        self.reverse_acceleration = Curve([(0.0, 1500)])
+        self.reverse_drifting_acceleration = Curve([(0.0, 0)])
+        self.braking_acceleration = Curve([(0.0, 1500)])
+        """
+        Acceleration force applied in opposition to car's forward speed whenever
+        neither gas nor brake are held.
+        """
+        self.max_engine_speed = 500
+        """
+        Engine is only strong enough to push car up to this speed.  It may be pushed higher by external forces,
+        but engine RPMs max out at this speed.
+        """
+        self.turning_radius_from_velocity = Curve([(0.0, 50.0), (600.0, 150.0)])
+        """
+        Steering speed is calculated from a turning radius which is derived from velocity.
+        This allows turning radius to increase as the car speeds up.
+        We use absolute speed in *any* direction so that the car continues to spin at an intuitive rate
+        even as you e.g. do a 720, drifting past enemies, shooting in all directions.  This is fun but
+        not realistic.
+        """
+        self.kinetic_friction = -10
+        """
+        Kinetic friction for wheels to resist lateral sliding.
+        Note: the math here is wonky, does not follow real-world physics rules.
+        """
+        self.static_friction = -100
+        "Static friction for wheels to resist lateral sliding."
+        self.lateral_velocity_kinetic_friction_threshold = 40
+        """
+        When lateral velocity exceeds this value, we switch from static to kinetic friction.
+        The goal is for drifting to start after you attempt to turn too aggressively.
+        TODO needs work.
+        """
+
+    def update(self, delta_time: float):
+        prev_velocity = self.velocity
+
+        # Holding both gas and reverse means you're in "drift mode"
+        drifting = (
+            self.input.accelerate_axis.value > 0 and self.input.brake_axis.value > 0
+        )
+        braking = (
+            self.input.accelerate_axis.value == 0 and self.input.brake_axis.value == 0
+        )
+
+        # Break down the car's current heading and velocity in a variety of ways.
```

Comment:
```suggestion
        # Break down the car's current facing and velocity in a variety of ways.
```
"heading is confusing